README.md

```diff
@@ -209,7 +209,85 @@ always read lower case. We can do:
 util.Try[String] = Success(foobar)
 ```
 
+In order for `pureconfig` to disambiguate between different options of a sealed
+family of case classes, it must read and write additional information in
+configurations. By default it uses the additional field `type`, encoding the
+concrete class represented in the configuration:
 
+```scala
+sealed trait AnimalConf
+case class DogConf(age: Int) extends AnimalConf
+case class BirdConf(canFly: Boolean) extends AnimalConf
+
+loadConfig[AnimalConf](parseString("""{ type: "dogconf", age: 4 }"""))
+// returns Success(DogConf(4))
+```
+
+For sealed families, `pureconfig` provides a way to customize the conversion
+without replacing the default `ConfigConvert`. By putting in scope an instance
+of `CoproductHint` for that sealed family, we can customize how the
+disambiguation is made. For example, if `type` clashes with one of the fields
+of a case class option, we can use another field:
+
+```scala
+implicit val animalConfHint = new FieldCoproductHint[AnimalConf]("kind")
+loadConfig[AnimalConf](parseString("""{ kind: "dogconf", age: 4 }"""))
+// returns Success(DogConf(4))
+```
+
+`FieldCoproductHint` can also be adapted to write class names in a different
+way:
+
+```scala
+implicit val animalConfHint = new FieldCoproductHint[AnimalConf]("type") {
+  override def fieldValue(name: String) = name.dropRight("Conf".length)
+}
+loadConfig[AnimalConf](parseString("""{ type: "Bird", canFly: true }"""))
+// returns Success(BirdConf(true))
+```
+
+With a `CoproductHint` you can even opt not to use any extra field at all. For
+example, if you encode enumerations using sealed traits, you can just write the
+name of the class:
+
+```scala
+import com.typesafe.config.{ConfigFactory,ConfigValue}
```

Comment:
It occurs to me that the main PR for `CoproductHint` didn't include a unit test for the  "enum-y" case. It seems like it might be useful to drop this code into a test to ensure it keeps working.